diag_manager/include/fms_diag_reduction_methods.inc

```diff
@@ -54,4 +54,107 @@ subroutine DO_TIME_NONE_ (data_out, data_in, mask, bounds_in, bounds_out, missin
     data_out(is_out:ie_out, js_out:je_out, ks_out:ke_out, :, 1) = missing_value
   end where
 
-end subroutine DO_TIME_NONE_
\ No newline at end of file
+end subroutine DO_TIME_NONE_
+
+!> @brief Do the time_min reduction method (i.e maintain the minimum value of the averaging time)
+subroutine DO_TIME_MIN_ (data_out, data_in, mask, bounds_in, bounds_out, missing_value)
+  real(FMS_TRM_KIND_),       intent(inout) :: data_out(:,:,:,:,:) !< output data
+  real(FMS_TRM_KIND_),       intent(in)    :: data_in(:,:,:,:)    !< data to update the buffer with
+  logical,                   intent(in)    :: mask(:,:,:,:)       !< mask
+  type(fmsDiagIbounds_type), intent(in)    :: bounds_in           !< indices indicating the correct portion
+                                                                  !! of the input buffer
+  type(fmsDiagIbounds_type), intent(in)    :: bounds_out          !< indices indicating the correct portion
+                                                                  !! of the output buffer
+  real(FMS_TRM_KIND_),       intent(in)    :: missing_value       !< Missing_value for data points that are masked
+
+  integer :: is_in, ie_in, js_in, je_in, ks_in, ke_in       !< Starting and ending indices of each dimention for
+                                                            !! the input buffer
+  integer :: is_out, ie_out, js_out, je_out, ks_out, ke_out !< Starting and ending indices of each dimention for
+                                                            !! the output buffer
+
+  integer :: i, j, k, l !< For looping
+
+  is_out = bounds_out%get_imin()
+  ie_out = bounds_out%get_imax()
+  js_out = bounds_out%get_jmin()
+  je_out = bounds_out%get_jmax()
+  ks_out = bounds_out%get_kmin()
+  ke_out = bounds_out%get_kmax()
+
+  is_in = bounds_in%get_imin()
+  ie_in = bounds_in%get_imax()
+  js_in = bounds_in%get_jmin()
+  je_in = bounds_in%get_jmax()
+  ks_in = bounds_in%get_kmin()
+  ke_in = bounds_in%get_kmax()
+
+  do i = 0, ie_out - is_out
+    do j = 0, je_out - js_out
+      do k = 0, ke_out - ks_out
+        do l = 0, size(data_out, 4) - 1
```

Comment:
Is this backwards?

Reply:
should it be l, k, j, i?

Reply:
https://github.com/NOAA-GFDL/FMS/pull/1367/commits/4193a084aba6e5b6a2535bddef0928b49db0ddb2